tests/test.cpp

```diff
@@ -719,6 +719,46 @@ void JsonEnumsTest() {
   TEST_EQ(std::string::npos != future_json.find("color: 13"), true);
 }
 
+void JsonOptionalTest() {
+  // load FlatBuffer schema (.fbs) and JSON from disk
+  std::string schemafile;
+  std::string jsonfile;
+  TEST_EQ(
+      flatbuffers::LoadFile((test_data_path + "optional_scalars.fbs").c_str(),
+                            false, &schemafile),
+      true);
+  TEST_EQ(
+      flatbuffers::LoadFile((test_data_path + "optional_scalars.json").c_str(),
+                            false, &jsonfile),
+      true);
+
+  auto include_test_path =
+      flatbuffers::ConCatPathFileName(test_data_path, "include_test");
+  const char *include_directories[] = { test_data_path.c_str(),
+                                        include_test_path.c_str(), nullptr };
+
+  // parse schema first, so we can use it to parse the data after
+  flatbuffers::Parser parser;
+  parser.opts.output_default_scalars_in_json = true;
```

Comment:
Can we also add a case for when this is false?

Reply:
Good idea, added it.